powersimdata/input/check.py

```diff
@@ -0,0 +1,195 @@
+    """
+    txfmr_branch_types = {"Transformer", "TransformerWinding"}
+    branch = grid.branch
+    bus2sub = grid.bus2sub
+    transformers = branch.loc[branch.branch_device_type.isin(txfmr_branch_types)].copy()
+    transformers["from_sub"] = bus2sub.loc[transformers.from_bus_id, "sub_id"].tolist()
+    transformers["to_sub"] = bus2sub.loc[transformers.to_bus_id, "sub_id"].tolist()
+    transformer_sub_match = transformers["from_sub"] == transformers["to_sub"]
+    if not transformer_sub_match.all():
+        non_matching_transformers = transformer_sub_match.index[~transformer_sub_match]
+        raise ValueError(
+            "Transformer(s) connected across multiple substations: "
+            f"{non_matching_transformers}."
+        )
+
+
+def _check_line_voltages(grid):
+    """Check whether any lines connect across difference voltage levels.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if any lines connect across difference voltage levels.
+    """
+    lines = grid.branch.query("branch_device_type == 'Line'").copy()
+    lines["from_kV"] = grid.bus.loc[lines.from_bus_id, "baseKV"].tolist()
+    lines["to_kV"] = grid.bus.loc[lines.to_bus_id, "baseKV"].tolist()
+    line_voltage_match = lines["from_kV"] == lines["to_kV"]
+    if not line_voltage_match.all():
+        non_matching_voltages = line_voltage_match.index[~line_voltage_match]
+        raise ValueError(
+            f"Lines(s) connected across multiple voltages: {non_matching_voltages}."
+        )
+
+
+def _check_plant_against_gencost(grid):
+    """Check whether indices of plant and gencost tables match.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if indices of plant and gencost tables don't match.
+    """
+    if not (
+        set(grid.plant.index) == set(grid.gencost["before"].index)
+        and set(grid.plant.index) == set(grid.gencost["after"].index)
+    ):
+        raise ValueError("indices for plant and gencost don't match")
+
+
+def _check_connected_components(grid):
+    """Check whether connected components and listed interconnects of a grid match.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if connected components and listed interconnects of a grid
+        don't match.
+    """
+    g = nx.from_pandas_edgelist(grid.branch, "from_bus_id", "to_bus_id")
+    num_connected_components = len([c for c in nx.connected_components(g)])
+    if len(grid.interconnect) == 1:
+        # Check for e.g. ['USA'] interconnect, which is really three interconnects
+        interconnect = grid.interconnect[0]
+        num_interconnects = (
+            grid.model_immutables.zones["interconnect_combinations"][interconnect]
+            if interconnect in grid.model_immutables.zones["interconnect_combinations"]
+            else 1
+        )
```

Comment:
This should work equivalently:
```python
num_interconnects = grid.model_immutables.zones["interconnect_combinations"].get(interconnect, 1)
```

Reply:
Not if we change `grid.model_immutables.zones["interconnect_combinations"]` to produce the set of interconnections, rather than the number.

Reply:
Then we could do 
```python
num_interconnects = len(grid.model_immutables.zones["interconnect_combinations"].get(interconnect, [dummy]))
```
instead?

Reply:
We could, but I don't like using `[dummy]` here.